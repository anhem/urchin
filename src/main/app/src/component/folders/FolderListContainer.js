import React, {Component} from 'react'
import {connect} from 'react-redux'
import {getFolders} from '../../action/folderAction'
import FolderList from "./FolderList";

export class FolderListContainer extends Component {

    componentWillMount() {
        this.props.getFolders();
    }

    render() {
        return (
            <FolderList
                folders={this.props.folders}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getFolders: () => {
            dispatch(getFolders())
        }
    }
};

const mapStateToProps = (state) => {
    return {
        folders: state.folderData.folders || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(FolderListContainer)