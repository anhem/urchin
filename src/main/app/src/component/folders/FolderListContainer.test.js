import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {FolderListContainer} from "./FolderListContainer";

describe('FolderListContainer', () => {

    let props = {
        getFolders: jest.fn(),
    };

    it('renders without crashing', () => {
        expect(mount(<FolderListContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<FolderListContainer {...props}/>))).toMatchSnapshot();
    });
});
