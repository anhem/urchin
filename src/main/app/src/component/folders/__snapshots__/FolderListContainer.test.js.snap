// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderListContainer match snapshot 1`] = `<FolderList />`;
